### PR TITLE
CNV-36964 - Updating mitigation info for VMStorageClassWarning

### PR DESCRIPTION
**Pull request by jherrman**
**What this PR does / why we need it**:

Fixes https://issues.redhat.com/browse/CNV-36964

### CODE REVIEW

**Comment by sradco** [comment]
/lgtm

@machadovilaca please review

**Comment by machadovilaca** [comment]
@jherrman please take a look at fix the linter failures

File: docs/deprecated_runbooks/VirtualMachineCRCErrors.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # VirtualMachineCRCErrors
 
+<!-- This file is deprecated, see VMStorageClassWarning for the relevant and up-to-date runbook content -->
```
Comment (sradco) [inline docs/deprecated_runbooks/VirtualMachineCRCErrors.md:3]: I believe this comment is relevant also to upstream and should be visible

Comment (sradco) [reply docs/deprecated_runbooks/VirtualMachineCRCErrors.md:3]: Hi @jherrman, WDYT?

Comment (jherrman) [reply docs/deprecated_runbooks/VirtualMachineCRCErrors.md]: Agreed, changed the line to be explicit.

**Comment by jherrman** [comment]
@sradco @machadovilaca , linter and other formalities should be taken care of. Please check again when convenient and let me know if anything else should be done before the changes can be merged.

Thanks!
J.

**Comment by sradco** [comment]
@jherrman can you please merge the 3 commits to one?

**Comment by jherrman** [comment]
> @jherrman can you please merge the 3 commits to one?

@sradco done :-)

**Review by sradco — APPROVED** [review]

**Comment by machadovilaca** [comment]
@jherrman in https://issues.redhat.com/browse/CNV-36964, it reads that:

> Documentation about mitigation makes it seem that you can just add the rxbounce option to the storage class and fix the CRC errors.  However, you cannot do that and a new storage class would need to be created with the new options added. 

but this issue was solved before, if you notice, the instructions you put between the upstream-only tags state that "It is recommended to create a dedicated StorageClass", not to edit any existing storage class anymore.

**Comment by jherrman** [comment]
@machadovilaca , true, it was fixed in the upstream version, but the downstream one still displays the problematic option config: https://github.com/kubevirt/monitoring/blob/main/docs/runbooks/VMStorageClassWarning.md#mitigation

So to amend that, I added a downstream-specific section to the mitigation. Would that work, or is that not a good way to go about dealing with the issue?

**Comment by machadovilaca** [comment]
It is just not updated in downstream, there were some issues with deprecated alerts, and we had to disable the automation

If we updated there with the current content of this upstream runbook that would solve the issue, right?

**Comment by jherrman** [comment]
> It is just not updated in downstream, there were some issues with deprecated alerts, and we had to disable the automation
> 
> If we updated there with the current content of this upstream runbook that would solve the issue, right?

@machadovilaca , if you mean current content as in the version I propose in this PR, then it definitely should. 

If you mean the version in https://github.com/kubevirt/monitoring/blob/main/docs/runbooks/VMStorageClassWarning.md , that one does not include links to internal RH resources, which would be a preferable solution for downstream purposes. 

Or is that not possible due to the sync automation issues?

**Comment by machadovilaca** [comment]
I agree, but that should be in the same format as for example: https://github.com/kubevirt/monitoring/blob/a0dcb2743e74a38a1059b50a584ca7207de06e39/docs/runbooks/CDIDataImportCronOutdated.md?plain=1#L105

I think we should keep the section you put inside `<!--USstart-->` as it was, and we should revert the changes in `docs/deprecated_runbooks/VirtualMachineCRCErrors.md`, we don't need that upstream

**Comment by jherrman** [comment]
> I agree, but that should be in the same format as for example:
> 
> https://github.com/kubevirt/monitoring/blob/a0dcb2743e74a38a1059b50a584ca7207de06e39/docs/runbooks/CDIDataImportCronOutdated.md?plain=1#L105

Wouldn't the `<!-- DS: text -->` syntax work only for a single line? Or is it somehow functionally problematic to use `<!--DSstart --> <!--DSend -->` instead ?

> 
> I think we should keep the section you put inside `<!--USstart-->` as it was, 

You mean removing the added upstream-only conditional tags? That would make the solution display twice (and in a somewhat conflicting form) in the downstream.

> and we should revert the changes in `docs/deprecated_runbooks/VirtualMachineCRCErrors.md`, we don't need that upstream

@sradco was of the opposite opinion - https://github.com/kubevirt/monitoring/pull/256#discussion_r1673951905

As for myself, provided that the "redirect" to the up-to-date runbook is accurate, I'd say it would improve the UX of the doc, rather than hamper it, no?

Cheers,
J.

**Comment by machadovilaca** [comment]
@jherrman upstream and downstream versions of this runbook are finally stable and in sync. Do you mind taking a new look at https://github.com/openshift/runbooks/blob/master/alerts/openshift-virtualization-operator/VMStorageClassWarning.md and see what changes are still needed?

**Comment by jherrman** [comment]
@machadovilaca , thank you for the heads-up! https://github.com/openshift/runbooks/blob/master/alerts/openshift-virtualization-operator/VMStorageClassWarning.md does look a lot better now - maybe apart from some minor wording/style issues, but those are not the objective of https://issues.redhat.com/browse/CNV-36964.

Therefore, I think it's safe to go ahead and close this PR (as well as the ticket).

Thanks again, and have an excellent day!
J.

**Pull request closed without merging**